src/fibsem_tools/io/fibsem_h5.py

```diff
@@ -0,0 +1,314 @@
+        data = FIBSEMData(np.rollaxis(dataset[:], 2), header)
+    return data
+
+
+def create_aggregate_fibsem_h5_file(
+    h5_filename: str,
+    dataset_names: Sequence[str],
+    data: Sequence[FIBSEMData],
+    raw_headers: Sequence[bytes] = None,
+    force: bool = False,
+    **kwargs
+):
+    """
+    Create aggregate FIBSEM HDF5 File
+
+    :param h5_filename: Filename of the HDF5 file to create.
+    :param dataset_names: Names of the datasets to create
+    :param data: A sequence of FIBSEMData instances.
+    :param raw_headers: Raw headers in the dat file.
+    :type raw_headers: bytes, optional
+    :param force: Overrwrite existing HDF5 file if True. defaults to False.
+    :type force: bool, optional
+
+    :returns: h5_filename
+    """
+    # Use 'x' permissions so we do not overwrite files
+    mode = "w" if force else "x"
+    f = h5py.File(h5_filename, mode)
+    if raw_headers is not None:
+        for n, d, h in zip(dataset_names, data, raw_headers):
+            dataset = create_fibsem_h5_dataset(f, n, d, **kwargs)
+            dataset.attrs["RawHeader"] = np.frombuffer(h, dtype="u1")
+    else:
+        for n, d in zip(dataset_names, data):
+            dataset = create_fibsem_h5_dataset(f, n, d, **kwargs)
+    f.close()
+    return h5_filename
+
+
+def create_aggregate_fibsem_h5_file_from_dat_filenames(
+    dat_filenames: Sequence[str], h5_dirname: str = None, **kwargs
+):
+    """
+    Create aggregate FIBSEM HDF5 File from .dat filenames
+
+    :param dat_filenames: A sequence of dat filenames
+    :param h5_filename: Name of HDF5 FILE. defaults to same directory as dat_filenames[0]
+    :type h5_filename: str, optional
+
+    :returns: h5_filename
+    """
+    prefix_name = ""
+    dataset_names = []
+    raw_headers = []
+
+    # Parse filenames of the form prefix_0_1_2.dat
+    r_dat_filename = re.compile(r"(.*)_(\d+-\d+-\d+).dat")
+    for dat_filename in dat_filenames:
+        dat_filename_base = os.path.basename(dat_filename)
+        m = re.fullmatch(r_dat_filename, dat_filename_base)
+        dataset_names.append(m[2])
+        raw_headers.append(_extract_raw_header(dat_filename))
+        if prefix_name == "":
+            prefix_name = m[1]
+        else:
+            # Ensure that all dat files have the same prefix
+            assert prefix_name == m[1], "All arguments must share a common prefix"
+
+    data = read_fibsem(dat_filenames)
+    if h5_dirname is None:
+        h5_dirname = os.path.dirname(dat_filenames[0])
+    h5_filename = os.path.join(h5_dirname, prefix_name + ".h5")
+
+    return create_aggregate_fibsem_h5_file(
+        h5_filename, dataset_names, data, raw_headers, **kwargs
+    )
+
+
+def create_default_h5_files(source_data, **kwargs):
+    """
+    Create HDF5 files with default settings
+
+    This is a high-level convenience function.
+
+    :param source_data: Source data. .dat filenames, a directory name, or a glob
```

Comment:
Let's not overload this initial function. I would prefer it if the first argument is a list of filenames which the calling code is responsible for creating. This function probably shouldn't be responsible for building that list via glob.

Reply:
I think it would be easier if this were a list of files that the code ingests which is what `create_aggregate_fibsem_h5_file_from_dat_filenames` currently takes. However, there the files must have the same prefix.

The convenience here is that you can give this a set of unrelated dat files and then it groups them into aggregate datasets. Thus you can just provide `*.dat`.

For what you want, I would just use `create_aggregate_fibsem_h5_file_from_dat_filenames`.